Issue 4548 - CLI - dsconf needs better root DN access control plugin …

mreynolds389 opened the pull request:
…validation

Description:  

There is no validation done for any of the root DN access control plugin settings.

Relates: https://github.com/389ds/389-ds-base/issues/4548

Reviewed by: ?

droideck commented on src/lib389/lib389/cli_conf/plugins/rootdn_ac.py:
```diff
@@ -20,8 +22,60 @@
+        try:
+            int(args.open_time)
+            if len(args.open_time) != 4:
+                raise ValueError("The open time must be a 4 digit number: HHMM")
```
As we're specifying the format, I'm thinking, maybe we should validate it accordingly? Like HH is 00-23 and MM is 00-59

mreynolds389 replied:
fixed

mreynolds389 committed:
Issue 4548 - CLI - dsconf needs better root DN access control plugin validation

Description:  There is no validation done for any of the root DN access control
              plugin settings.

Relates: https://github.com/389ds/389-ds-base/issues/4548

Reviewed by: spichugi & firstyear (Thanks!!)

droideck commented on src/lib389/lib389/cli_conf/plugins/rootdn_ac.py:
```diff
@@ -20,8 +22,71 @@
+            raise ValueError(f"The hour portion of the time is invalid: {hour}  Must be between 0 and 23")
+        min = int(args.open_time[-2:])
+        if min < 1 or min > 59:
+            raise ValueError(f"The minute portion of the time is invalid: {min}  Must be between 1 and 59")
```
Maybe I miss something but I think it should allow 0 to 59:

```
$ dsconf localhost plugin root-dn set --close-time 0100
Error: The minute portion of the time is invalid: 0  Must be between 1 and 59
```

mreynolds389 replied:
Heh, you're right :-)  I'll get that changed and merged...

droideck reviewed (approved):
Besides the very small issue - you have my ack!
Feel free to merge:)
